interfaces/prompting/requestrules/requestrules.go

```diff
@@ -653,6 +653,11 @@ func (rdb *RuleDB) populateNewRule(user uint32, snap string, iface string, const
 		Lifespan:    lifespan,
 		Expiration:  expiration,
 	}
+
+	if err := newRule.validate(); err != nil {
+		return nil, err
+	}
```

Comment:
Nice! `lifespan.ParseDuration()` checks all the things `rule.validate()` does, so the latter isn't really necessary, but can't hurt I suppose.

Reply:
Perhaps the other checks above should be added to `rule.validate()` as well, so they're run on `load()`?
- `rule.Constraints.ValidateForInterface(rule.Interface)`
- `rule.Outcome.AsBool()`
- `rule.Lifespan != prompting.LifespanSingle`
- `rule.Lifespan.ValidateExpiration(rule.Expiration, currTime)`

This would simplify `load()` as well. In this case, I think we'd pass `time.Now()` as `currTime` into `rule.validate()`.